metadata-ingestion/src/datahub/masking/secret_registry.py

```diff
@@ -0,0 +1,461 @@
+#   - If unsure, don't add it - users can customize via skip list/pattern
+#   - Better to mask too much than too little
+SYSTEM_ENV_VARS: Set[str] = {
+    # Paths
+    "HOME",
+    "PWD",
+    "OLDPWD",
+    "PATH",
+    "TMPDIR",
+    "TEMP",
+    "TMP",
+    "VIRTUAL_ENV",
+    # Shell and Terminal
+    "SHELL",
+    "TERM",
+    "TERM_PROGRAM",
+    "TERM_PROGRAM_VERSION",
+    "TERM_SESSION_ID",
+    "COLORTERM",
+    "CLICOLOR",
+    "CLICOLOR_FORCE",
+    # Locale and Language
+    "LANG",
+    "LANGUAGE",
+    "LC_ALL",
+    "LC_CTYPE",
+    "LC_MESSAGES",
+    # Display and Graphics
+    "DISPLAY",
+    "WAYLAND_DISPLAY",
+    # User Info
+    "USER",
+    "USERNAME",
+    "LOGNAME",
+    "UID",
+    # System Info
+    "HOSTNAME",
+    "HOSTTYPE",
+    "OSTYPE",
+    "MACHTYPE",
+    # Development Tools
+    "EDITOR",
+    "VISUAL",
+    "PAGER",
+    "LESS",
+    "GIT_EDITOR",
+    # Python-specific
+    "PYTHONPATH",
+    "PYTHONHOME",
+    "PYTHONIOENCODING",
+    "PYTHONUNBUFFERED",
+    "PYTHONDONTWRITEBYTECODE",
+    # Java
+    "JAVA_HOME",
+    "JAVA_OPTS",
+    # Node/NPM
+    "NODE_ENV",
+    "NVM_DIR",
+    "NPM_CONFIG_PREFIX",
+    # Build Tools
+    "GRADLE_HOME",
+    "MAVEN_HOME",
+    "M2_HOME",
+    # Package Managers
+    "HOMEBREW_PREFIX",
+    "HOMEBREW_CELLAR",
+    "HOMEBREW_REPOSITORY",
+    # Version Managers
+    "ASDF_DIR",
+    "ASDF_DATA_DIR",
+    "PYENV_ROOT",
+    "RBENV_ROOT",
+    # Color/Display Settings
+    "LSCOLORS",
+    "LS_COLORS",
+    "GREP_COLOR",
+    "GREP_COLORS",
+    # Misc System
+    "SHLVL",
+    "PS1",
+    "PS2",
+    "IFS",
+    "MANPATH",
+    "INFOPATH",
+    # macOS Specific
+    "__CF_USER_TEXT_ENCODING",
+    "__CFBundleIdentifier",
+    "XPC_FLAGS",
+    "XPC_SERVICE_NAME",
+    # SSH (paths only, not keys)
+    "SSH_AUTH_SOCK",  # Socket path, not a secret
+    # ZSH
+    "ZSH",
+    "ZSH_VERSION",
+    # DataHub Debug (not secrets)
+    "DATAHUB_DEBUG",
+    "DATAHUB_LOG_LEVEL",
+    "DATAHUB_DISABLE_SECRET_MASKING",
+    "DATAHUB_MASKING_ENV_VARS_SKIP_LIST",
+    "DATAHUB_MASKING_ENV_VARS_SKIP_PATTERN",
+    # Claude Code
+    "CLAUDE_CODE_ENTRYPOINT",
+    # Telemetry
+    "OTEL_EXPORTER_OTLP_ENDPOINT",
+}
+
+
+def should_mask_env_var(var_name: str) -> bool:
+    """
+    Determine if an environment variable should be masked.
+
+    Filtering strategy:
+    1. Skip if in SYSTEM_ENV_VARS (hardcoded common variables)
+    2. Skip if in user skip list (DATAHUB_MASKING_ENV_VARS_SKIP_LIST)
+    3. Skip if matches user skip pattern (DATAHUB_MASKING_ENV_VARS_SKIP_PATTERN)
+    4. Mask all other variables
+
+    Args:
+        var_name: Name of the environment variable
+
+    Returns:
+        True if the variable should be masked, False otherwise
+    """
+    # 1. Check hardcoded system variables
+    if var_name in SYSTEM_ENV_VARS:
+        return False
+
+    # 2. Check user-provided skip list
+    skip_list_env = os.getenv("DATAHUB_MASKING_ENV_VARS_SKIP_LIST", "")
```

Comment:
we can move these two `os.getenv` to the `env_vars.py` file and return proper string list and regexp respectively